runtimes/runtimes/lsp/router/util.ts

```diff
@@ -25,11 +25,13 @@ export function asPromise(value: any): Promise<any> {
 }
 
 export function mergeObjects(obj1: any, obj2: any) {
```

Comment:
seems like maintaining this function can be brittle. 
Might be worth considering replacing it with lodash's merge - https://lodash.com/docs/4.17.15#merge

https://www.npmjs.com/package/lodash

Reply:
interesting, skimming the details it does appear to work for mixed data structs and does so recursively